js/post/update_post.js

```diff
@@ -5,200 +5,165 @@ var completed_spec = {'CSC108': 0, 'CSC148': 0, 'CSC165': 0, 'Calc1': 0, 'CSC207
     }
 
-    // Update Extra
-
-    countMaj = 0; 
-    countSpec = 0;
-    var specExtra = $('#spec_extra')[0].getElementsByTagName('input');
-    var majExtra = $('#maj_extra')[0].getElementsByTagName('input');
-    for (var l = 0; l < 4; l++) {
-        if (specExtra[l].value != '') {
-            countSpec += 1;
-        } if (l < 3) {
-            if (majExtra[l].value != '') {
-                countMaj += 1;
-            }
-        }
+    if (filled_textboxes_400['maj'] === 1) {
+        updateCategory($('#maj_400')[0].getElementsByClassName('code')[0], 'fulfilled');
+        categories_completed['maj'] += 1;
+    } else {    
+        updateCategory($('#maj_400')[0].getElementsByClassName('code')[0], 'not fulfilled'); 
     }
 
-    if (countSpec === 4) {
+    // Update Extra
+    if (filled_textboxes_extra['spec'] === 4) {
         updateCategory($('#spec_extra')[0].getElementsByClassName('code')[0], 'fulfilled');
-    } if (countMaj === 3) {
+        categories_completed['spec'] += 1;
+    } if (filled_textboxes_extra['maj'] === 3) {
         updateCategory($('#maj_extra')[0].getElementsByClassName('code')[0], 'fulfilled');
-    } if (countSpec < 4) {
+        categories_completed['maj'] += 1;
+    } if ((filled_textboxes_300['min'] + filled_textboxes_400['min'] + filled_textboxes_200) === 3) {
+        updateCategory($('#min_misc')[0].getElementsByClassName('code')[0], 'fulfilled');
+        categories_completed['min'] += 1;
+    } if (filled_textboxes_extra['spec'] < 4) {
         updateCategory($('#spec_extra')[0].getElementsByClassName('code')[0], 'not fulfilled');
-    } if (countMaj < 3) {
+    } if (filled_textboxes_extra['maj'] < 3) {
         updateCategory($('#maj_extra')[0].getElementsByClassName('code')[0], 'not fulfilled');
+    } if ((filled_textboxes_300['min'] + filled_textboxes_400['min'] + filled_textboxes_200) < 3) {
+        updateCategory($('#min_misc')[0].getElementsByClassName('code')[0], 'not fulfilled');
     }
 
+    checkPostCompleted();
+}
+
+/**
+ * Resets all values to initial starting values
+**/
+function resetValues() {
+    'use strict';
+
+    index300 = {'spec': 0, 'maj': 0, 'min': 0};
+    index400 = {'spec': 0, 'maj': 0, 'min': 0};
+    categories_completed = {'spec': 0, 'maj': 0, 'min': 0};
+    index200 = 0;
+
+    creditCount300and400['maj'] = 0;
+    creditCount300and400['spec'] = 0;
+    creditCount300and400['min'] = 0;
+
+    filled_textboxes_300 = {'spec': 0, 'maj': 0, 'min': 0};
+    filled_textboxes_400 = {'spec': 0, 'maj': 0, 'min': 0};
+    filled_textboxes_extra = {'spec': 0, 'maj': 0, 'min': 0};
+    filled_textboxes_200 = 0;
+}
+
+
+/**
+ * Updates categories for required courses in each POSt
+ * @param {string} post The POSt that you are updating categories for.
+**/
+function updateReqsCategory(post) {
+    'use strict';
+
+    var array = 'completed_' + post;
+    for (var property in window[array]) {
```

Comment:
You declare property several times. Many variables are declared multiple times in this file.